include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
+// On older versions of Mac, the system's libc++ does not provide isnan(double)
+// So we just provide it ourselves.
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, bool>::type
```

Comment:
I don't quite understand this. Why do we need `isnan()` for integer types? And if we need it (e.g. for reasons of genericity), why don't we just return `true` then?